File: frontend/PKI-Certificates-Management-System/src/app/model/certificate.ts
import { end, start } from '@popperjs/core';

export class Certificate {
  issuer: string;
  subject: string;
  startDate: string;
  endDate: string;
  type: string;
  alias: string;
  isRevoked: boolean;

  constructor(
    issuer: string,
    subject: string,
    startDate: string,
    endDate: string,
    type: string,
    alias: string,
    isRevoked: boolean
  ) {
    this.issuer = issuer;
    this.subject = subject;
    this.startDate = startDate;
    this.endDate = endDate;
    this.type = type;
    this.alias = alias;
    this.isRevoked = isRevoked;
  }
}

File: backend/src/main/java/pki/certificates/management/model/Certificate.java
package pki.certificates.management.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.security.cert.X509Certificate;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
public class Certificate {

    private Subject subject;
    private Issuer issuer;
    private String serialNumber;
    private Date startDate;
    private Date endDate;

    // svi prethodni podaci mogu da se izvuku i iz X509Certificate, osim privatnog kljuca issuera
    private X509Certificate x509Certificate;

}
